neotron-bmc-commands/src/lib.rs

```diff
@@ -0,0 +1,174 @@
+	/// # UART Control
+	/// Settings for the UART
+	/// * Length: 1
+	/// * Mode: R/W
+	UartControl = 0x32,
+	/// # UART Status
+	/// The current state of the UART
+	/// * Length: 1
+	/// * Mode: R/W1C
+	UartStatus = 0x33,
+	/// # UART Baud Rate
+	/// The UART baud rate in bps, as a `u32le`
+	/// * Length: 4
+	/// * Mode: R/W
+	UartBaudRate = 0x34,
+	/// # PS/2 Keyboard Receive/Transmit Buffer
+	/// Data received/to be sent over the PS/2 keyboard port
+	/// * Length: up to 16
+	/// * Mode: FIFO
+	Ps2KbBuffer = 0x40,
+	/// # PS/2 Keyboard Control
+	/// Settings for the PS/2 Keyboard port
+	/// * Length: 1
+	/// * Mode: R/W
+	Ps2KbControl = 0x41,
+	/// # PS/2 Keyboard Status
+	/// Current state of the PS/2 Keyboard port
+	/// * Length: 1
+	/// * Mode: R/W1C
+	Ps2KbStatus = 0x42,
+	/// # PS/2 Mouse Receive/Transmit Buffer
+	/// Data received/to be sent over the PS/2 Mouse port
+	/// * Length: up to 16
+	/// * Mode: FIFO
+	Ps2MouseBuffer = 0x50,
+	/// # PS/2 Mouse Control
+	/// Settings for the PS/2 Mouse port
+	/// * Length: 1
+	/// * Mode: R/W
+	Ps2MouseControl = 0x51,
+	/// # PS/2 Mouse Status
+	/// Current state of the PS/2 Mouse port
+	/// * Length: 1
+	/// * Mode: R/W1C
+	Ps2MouseStatus = 0x52,
+	/// # I²C Receive/Transmit Buffer
+	/// Data received/to be sent over the I²C Bus
+	/// * Length: up to 16
+	/// * Mode: FIFO
+	I2cBuffer = 0x60,
+	/// # I²C FIFO Control
+	/// Settings for the I²C FIFO
+	/// * Length: 1
+	/// * Mode: R/W
+	I2cFifoControl = 0x61,
+	/// # I²C Control
+	/// Settings for the I²C Bus
+	/// * Length: 1
+	/// * Mode: R/W
+	I2cControl = 0x62,
+	/// # I²C Status
+	/// Current state of the I²C Bus
+	/// * Length: 1
+	/// * Mode: R/W1C
+	I2cStatus = 0x63,
+	/// # I²C Baud Rate
+	/// The I²C clock rate in Hz, as a `u32le`
+	/// * Length: 4
+	/// * Mode: R/W
+	I2cBaudRate = 0x64,
+}
+
+impl Command {
+	pub fn parse(byte: u8) -> Option<Command> {
+		match byte {
+			0x00 => Some(Command::ProtocolVersion),
+			0x01 => Some(Command::FirmwareVersion),
+			0x10 => Some(Command::InterruptStatus),
+			0x11 => Some(Command::InterruptControl),
+			0x20 => Some(Command::ButtonStatus),
+			0x21 => Some(Command::SystemTemperature),
+			0x22 => Some(Command::SystemVoltage33S),
+			0x23 => Some(Command::SystemVoltage33),
+			0x24 => Some(Command::SystemVoltage55),
+			0x25 => Some(Command::PowerControl),
+			0x30 => Some(Command::UartBuffer),
+			0x31 => Some(Command::UartFifoControl),
+			0x32 => Some(Command::UartControl),
+			0x33 => Some(Command::UartStatus),
+			0x34 => Some(Command::UartBaudRate),
+			0x40 => Some(Command::Ps2KbBuffer),
+			0x41 => Some(Command::Ps2KbControl),
+			0x42 => Some(Command::Ps2KbStatus),
+			0x50 => Some(Command::Ps2MouseBuffer),
+			0x51 => Some(Command::Ps2MouseControl),
+			0x52 => Some(Command::Ps2MouseStatus),
+			0x60 => Some(Command::I2cBuffer),
+			0x61 => Some(Command::I2cFifoControl),
+			0x62 => Some(Command::I2cControl),
+			0x63 => Some(Command::I2cStatus),
+			0x64 => Some(Command::I2cBaudRate),
```

Comment:
This looks a bit error prone. If you don't mind an additional dependency, something like [num_enum](https://docs.rs/num_enum/0.5.7/num_enum/) could be used to avoid writing the conversions manually.

Reply:
That's a good idea, thanks!

Reply:
It grinds my gears that you can't `match u8` and use a `repr(u8)` as the pattern match.

Reply:
I also wonder if I should just re-export the key codes from the `pc-keyboard` crate. Saves a big conversion table inside the BIOS.